### PR TITLE
Feat(frontend): Reat.JS Tutorial 

### LINKED ISSUE #270: React JS tutorial
Create the english version for the react tutorial 

Related to :  Tutorial React JS #183

### PR DESCRIPTION
**Pull request by steph222**
### GH Issue

Resolves #270  

### Steps to test
1. Go to Tutorials/React JS
2. Check the content is right

#### CheckList
- [ ] Follow proper Markdown format
- [ ] The content is adequate
- [ ] The content is equal to https://guias.eoscostarica.io/docs/tutoriales/tutorial-react

### CODE REVIEW

File: docs/tutorials/react-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+
+React is a JavaScript library that will be used to build web interfaces and mobile applications with an extension called [React Native](https://reactnative.dev/). React works with units of code that work independently. React also uses a tool called JSX that allows a very simple interaction with JavaScript, that is, it can also be used with HTML in a simple way.
+
+>[More Information](https://reactjs.org)  
+
+## Tools
+
+For React the following tools are suggested, tools with * are required:
+
+* Web Navigator (Google Chorme, Moxilla Firefox)*
```
Comment (xavier506) [inline docs/tutorials/react-tutorial.md:15]: Web Browser

File: docs/tutorials/react-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+* [Node Js](https://nodejs.org/en/)*
```
Comment (xavier506) [inline docs/tutorials/react-tutorial.md:16]: Node JS

File: docs/tutorials/react-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+    <div className="App">
+      <p>Hola Mundo</p>
+    </div>
+  );
+}
+}
+export default App;
+```
+
+## What are the Components?
```
Comment (xavier506) [inline docs/tutorials/react-tutorial.md:145]: What are Components?

File: docs/tutorials/react-tutorial.md [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+* The entire virtual DOM is updated.
+* Virtual DOM is compared to a version prior to upgrade. React notices which objects have changed.
+* The updated object and only that object is updated in the actual DOM.
+* Changes to the actual DOM cause the screen to change.
+
+>This process is known as reconciliation.
+
+* * *
+
+## Check Facilities
```
Comment (Edgar097) [inline docs/tutorials/react-tutorial.md:66]: Check `Versions` or `Installations` instead of Facilities, i think is better

**Review by Edgar097 — APPROVED** [review]

File: docs/tutorials/react-tutorial.md [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+
+### JavaScript Standards
+
+It is important to follow the JavaScript standards, it is recommended to use the [ESLint](https://eslint.org/) library to help you follow these [JavaScript standards](https://standardjs.com/rules.html).
+
+## Virtual DOM
+
+**Document Object Model for Document Representation** (Document Object Model) is essentially a platform interface that provides a standard set of objects to represent HTML, XHTML and XML documents. A standard model on how these objects can be combined, and a standard interface to access and manipulate them.
+
+React realized that updating the full screen is a task that can be costly in performance and time. So I create a Virtual DOM. This virtual DOM object is identical to the JavaScript DOM object.
```
Comment (leisterfrancisco) [inline docs/tutorials/react-tutorial.md:53]: Not sure about this sentence:
> So I create a Virtual DOM

**Review by leisterfrancisco — APPROVED** [review]
Good work 👍

File: docs/tutorials/react-tutorial.md [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          Learn React
+        </a>
+      </header>
+    </div>
+  );
+}
+export default App;
```
Comment (tetogomez) [inline docs/tutorials/react-tutorial.md:123]: add blank space before export

File: docs/tutorials/react-tutorial.md [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+```
+
+You can change the content to the following and make the changes you want as the first steps for your adventure in React. In addition, you can see the changes in your browser, for this you must keep your terminal with the server open.
+
+```javascript
+import { Component } from 'react';
+
+	import logo from './logo.svg';
```
Comment (tetogomez) [inline docs/tutorials/react-tutorial.md:131]: please, check the indentation

and add a blank space after the import group

File: docs/tutorials/react-tutorial.md [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+class App extends Component{
+  render(){
+  return (
+    <div className="App">
+      <p>Hola Mundo</p>
+    </div>
+  );
+}
+}
+export default App;
```
Comment (tetogomez) [inline docs/tutorials/react-tutorial.md:142]: Blank space before export

File: docs/tutorials/react-tutorial.md [hunk 9]
```diff
@@ -0,0 +1,299 @@
...
+
+## What are Components?
+
+Components are defined as a single unit of code that will have a specifically assigned responsibility that certain behaviors can be encapsulated in components.
+
+This is an example of a component. Create a new file inside your `components` folder with the name of `header.js` and in this way you can use it later. This type of component is no longer used in EOS Costa Rica, however it is important to understand its definition and use.
+
+```javascript
+import React, { Component } from "react";
+class Header extends Component{
```
Comment (tetogomez) [inline docs/tutorials/react-tutorial.md:153]: blank space

File: docs/tutorials/react-tutorial.md [hunk 10]
```diff
@@ -0,0 +1,299 @@
...
+    render(){
+        return(
+            <div>
+                <h1>Ejemplo </h1>
+            </div>
+        );
+    }
+}
+export default Header;
```
Comment (tetogomez) [inline docs/tutorials/react-tutorial.md:162]: same here

File: docs/tutorials/react-tutorial.md [hunk 11]
```diff
@@ -0,0 +1,299 @@
...
+This component can be used as follows inside the `App.js`:
+
+```javascript
+import { Component } from 'react';
+
+import './styles/App.css';
+
+import Header from "./components/header" //Se debe agregar el import del componente
+
+	class App extends Component{
```
Comment (tetogomez) [inline docs/tutorials/react-tutorial.md:174]: check identation

File: docs/tutorials/react-tutorial.md [hunk 12]
```diff
@@ -0,0 +1,299 @@
...
+  render(){
+  return (
+    <div className="App">
+      <Header />  /*Se agrega el componente*/
+
+    </div>
+  );
+}
+}
+export default App;
```
Comment (tetogomez) [inline docs/tutorials/react-tutorial.md:184]: blank space

File: docs/tutorials/react-tutorial.md [hunk 13]
```diff
@@ -0,0 +1,299 @@
...
+* [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo)
+* [useRef](https://reactjs.org/docs/hooks-reference.html#useref)
+* [useImperativeHandle](https://reactjs.org/docs/hooks-reference.html#useimperativehandle)
+* [useLayoutEffect](https://reactjs.org/docs/hooks-reference.html#uselayouteffect)
+* [useDebugValue](https://reactjs.org/docs/hooks-reference.html#usedebugvalue)
+
+This is an example of use, you can replace the code of your `App.js` to test it. In this example they show the dimensions of the screen even when it is resized.
+
+```javascript
+import React , { useState, useEffect } from "react";
```
Comment (tetogomez) [inline docs/tutorials/react-tutorial.md:264]: Add a blank space between import groups

example: 
```
import React , { useState, useEffect } from "react"

import "./styles/App.css"

const App = () => {...
```

File: docs/tutorials/react-tutorial.md [hunk 14]
```diff
@@ -0,0 +1,299 @@
...
+import "./styles/App.css"
+const App = () => {
+  const [width,setWidth] = useState(window.innerWidth);
+  const [height, setHeight]= useState(window.innerHeight);
+  const updateDimensions = () => {
+    setWidth(window.innerWidth);
+    setHeight(window.innerHeight);
+  }
```
Comment (tetogomez) [inline docs/tutorials/react-tutorial.md:272]: always add a blank space after const, let, or var declaration group

example: 
```
const App = () => {
  const [width,setWidth] = useState(window.innerWidth);
  const [height, setHeight]= useState(window.innerHeight);
  const updateDimensions = () => {
    setWidth(window.innerWidth);
    setHeight(window.innerHeight);
  }
  
  useEffect( () => {...
  ```

File: docs/tutorials/react-tutorial.md [hunk 15]
```diff
@@ -0,0 +1,299 @@
...
+  useEffect( () => {
+    window.addEventListener('resize', updateDimensions);
+    return() => {
```
Comment (tetogomez) [inline docs/tutorials/react-tutorial.md:275]: blank space before `return`

File: docs/tutorials/react-tutorial.md [hunk 16]
```diff
@@ -0,0 +1,299 @@
...
+      window.removeEventListener('resize',updateDimensions);
+    }
+  }
+  )
+  return(
```
Comment (tetogomez) [inline docs/tutorials/react-tutorial.md:280]: same here

File: docs/tutorials/react-tutorial.md [hunk 17]
```diff
@@ -0,0 +1,299 @@
...
+    <div className="App">
+      Width = {width}
+      <br/>
+      Height = {height}
+    </div>
+  );
+}
+export default App;
```
Comment (tetogomez) [inline docs/tutorials/react-tutorial.md:288]: and here

**Review by tetogomez — COMMENTED** [review]
check the comments

**Review by xavier506 — APPROVED** [review]
LGTM

**Review by tetogomez — APPROVED** [review]
LGTM

Nice job @steph222

**Pull request merged**